code/game/objects/structures/mouse_cage.dm

```diff
@@ -0,0 +1,108 @@
+
+/obj/item/critter_cage
+	name = "small cage"
+	desc = "A safe place where to keep tiny animals safe. Fit with a drinking bottle that can be refilled."
+	icon = 'icons/obj/virology.dmi'
+	icon_state = "cage"
+	density = 1
+	anchored = 0
+	pressure_resistance = 5
+	flags = FPRINT  | OPENCONTAINER
+
+	var/lock_type = /datum/locking_category/buckle/cage
+
+	var/mob/living/simple_animal/critter = null
+
+
+/obj/item/critter_cage/New()
+	..()
+	create_reagents(10)
+
+/obj/item/critter_cage/Destroy()
+	if (critter)
+		unlock_atom(critter)
+		critter.pixel_x = 0
+		critter.pixel_y = 0
+		critter = null
+	..()
+
+/datum/locking_category/buckle/cage
+
+/obj/item/critter_cage/attack_paw(var/mob/user)
+	return attack_hand(user)
+
+/obj/item/critter_cage/attack_hand(var/mob/user)
+	if (critter)
+		if( !user.get_active_hand() )
+			unlock_atom(critter)
+			critter.pixel_x = 0
+			critter.pixel_y = 0
+			critter.scoop_up(user)
+			user.visible_message("<span class='notice'>[user] picks up \the [critter].</span>", "<span class='notice'>You pick up \the [critter].</span>")
+			critter = null
+	else
+		MouseDropFrom(user)
+
+/obj/item/critter_cage/attackby(var/obj/O,var/mob/user)
+	. = ..()
+
+	if (!critter && istype (O,/obj/item/weapon/holder/animal))
+		var/obj/item/weapon/holder/animal/store = O
+		var/mob/living/simple_animal/inside = store.stored_mob
+		if (inside.size > SIZE_TINY)
+			to_chat(user, "<span class='warning'>\The [inside] is too big for \the [src]!</span>")
+			return
+		if(!user.drop_item(O, loc))
+			to_chat(user, "<span class='warning'>You can't let go of \the [O]!</span>")
+			return
+		critter = inside
+		qdel(store)
+		critter.forceMove(loc)
+		critter.pixel_x = pixel_x
```

Comment:
pixel-shifting should be done inside an overridden lock_atom and unlock_atom, rather than before each time you call one of those